meerk40t/gui/scene/guicolors.py

```diff
@@ -0,0 +1,174 @@
+            wx.Colour(0xA0, 0xA0, 0xA0, 128),
+            wx.Colour(0x00, 0x00, 0x00),
+            wx.Colour("Grey"),
+            wx.Colour(0xFF, 0xA0, 0xA0, 0x60),
+            wx.Colour(0xA0, 0xA0, 0xA0, 0x40),
+            wx.Colour(0x00, 0xFF, 0x00, 0xA0),
+            wx.Colour(0xFF, 0x00, 0x00),
+            wx.Colour(0x00, 0xFF, 0x00),
+            wx.Colour(0x00, 0x00, 0xFF),
+            wx.Colour(0xFF, 0xFF, 0xFF)
+        )
+
+        self.context.setting(str, "color_manipulation", color_to_str(self.default_color[0].GetRGBA()))
+        self.context.setting(str, "color_laserpath", color_to_str(self.default_color[1].GetRGBA()))
+        self.context.setting(str, "color_grid", color_to_str(self.default_color[2].GetRGBA()))
+        self.context.setting(str, "color_guide", color_to_str(self.default_color[3].GetRGBA()))
+        self.context.setting(str, "color_background", color_to_str(self.default_color[4].GetRGBA()))
+        self.context.setting(str, "color_magnetline", color_to_str(self.default_color[5].GetRGBA()))
+        self.context.setting(str, "color_snap_visible", color_to_str(self.default_color[6].GetRGBA()))
+        self.context.setting(str, "color_snap_closeup", color_to_str(self.default_color[7].GetRGBA()))
+        self.context.setting(str, "color_selection1", color_to_str(self.default_color[8].GetRGBA()))
+        self.context.setting(str, "color_selection2", color_to_str(self.default_color[9].GetRGBA()))
+        self.context.setting(str, "color_selection3", color_to_str(self.default_color[10].GetRGBA()))
+        self.context.setting(str, "color_bed", color_to_str(self.default_color[11].GetRGBA()))
+
+    def set_default_colors(self):
+        '''
+        Reset all colors to default values...
+        '''
+        self.context("set color_manipulation %s" % color_to_str(self.default_color[0].GetRGBA()))
+        self.context("set color_laserpath %s" % color_to_str(self.default_color[1].GetRGBA()))
+        self.context("set color_grid %s" % color_to_str(self.default_color[2].GetRGBA()))
+        self.context("set color_guide %s" % color_to_str(self.default_color[3].GetRGBA()))
+        self.context("set color_background %s" % color_to_str(self.default_color[4].GetRGBA()))
+        self.context("set color_magnetline %s" % color_to_str(self.default_color[5].GetRGBA()))
+        self.context("set color_snap_visible %s" % color_to_str(self.default_color[6].GetRGBA()))
+        self.context("set color_snap_closeup %s" % color_to_str(self.default_color[7].GetRGBA()))
+        self.context("set color_selection1 %s" % color_to_str(self.default_color[8].GetRGBA()))
+        self.context("set color_selection2 %s" % color_to_str(self.default_color[9].GetRGBA()))
+        self.context("set color_selection3 %s" % color_to_str(self.default_color[10].GetRGBA()))
+        self.context("set color_bed %s" % color_to_str(self.default_color[11].GetRGBA()))
+
+    @property
+    def color_manipulation(self):
```

Comment:
Making these properties might not matter if it was a context you should be able to access that from any other context. `self.context.get_context("colors")` for example. Then .manipulation would have that color etc. And such things could be decentralized. So nothing needs to access a central color location. Just add your color to the color context and do so within the code doing the accessing. Not that there's anything that wrong doing it this way.